Added Role UI BDD feature file and implementations

miguelcleverti opened the pull request:
RoleUI BDD sentences and implentations cherry-picked and squashed from #296

miguelcleverti commented:
@andrerom @pcardiga @joaoinacio

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,167 @@
+    Then I am notified that the name already exists
+
+  @javascript @common
+  Scenario: cancel the update of a role
```
_C_ancel :)

andrerom commented on Resources/views/Role/list_roles.html.twig:
```diff
@@ -30,10 +30,10 @@
+                        <tr class="ez-role">
+                            <td class="ez-role-name"><a href="{{ path("admin_roleView", {"roleId": role.id}) }}">{{ role.identifier }}</a></td>
+                            <td class="ez-role-id">{{ role.id }}</td>
+                            <td class="ez-role-edit-button">
```
Looks ok with me, but maybe @dpobel has a POV on how we should sugar the output to be testable.

dpobel replied:
well, a class called `button` on something which is absolutely not a button, that's not good. You can add such class on the `a` but by following the naming convention of PlatformUI, ie `ez-button-role-edit`

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,167 @@
+    And I should see a "Cancel" button
+
+  @javascript @common
+  Scenario: create a valid role
```
**C**reate, also below :)

andrerom commented:
Besides nitpicks, and possibly input from DP on how we semantically sugar the html output for testing, +1

miguelcleverti commented:
@andrerom nitpicks fixed :smile:

andrerom commented:
passing and ready for merge? :)

miguelcleverti commented:
@andrerom only the delete is not passing, because it is not possible to delete role yet

miguelcleverti commented:
@andrerom after all it is possible to delete roles role, will need just some extra implementation so no merge for now

pcardiga commented:
The possibility of deleting roles was introduced by latest implementation on Role UI

andrerom commented on Features/Context/SubContext/CommonActions.php:
```diff
@@ -184,14 +184,17 @@ protected function clickElementByText($text, $selector, $textSelector = null, $i
      * @param int       $iteration      Iteration number, used to control number of executions
      * @return array
      */
     protected function getElementByText($text, $selector, $textSelector = null, $iteration = 3)
```
baseElement missing

miguelcleverti replied:
I noticed, strange it's not there :neutral_face:

miguelcleverti commented:
Implemented the delete sentences, fixed all the issues caused by the role versioning and removed the "button" from the class name. All tests are passing, ready for merge @andrerom

andrerom commented on Features/Context/PlatformUI.php:
```diff
@@ -78,6 +79,7 @@ protected function waitForLoadings()
         while ($page->find('css', $loadingSelector) != null) {
             usleep(100 * 1000); // 100ms
         }
+        sleep(1);
```
I din't quite get why this is needed, it seem to already be matched above, and would another `usleep(100 * 1000)` be enough?

andrerom replied:
either way, need a inline comment here to mention why this is here.

miguelcleverti replied:
I decided to leave this here because this way it does not happen the "random" race condition errors, this is more of a temporary solution while we don't find a better/definitive one, but I forgot to had some comments about it @andrerom

andrerom commented:
+1 awaiting checks before merge
